test/unit/TestParser.ml

```diff
@@ -0,0 +1,41 @@
+open Lib
+
+let test_exists_bigger () =
+  Alcotest.(check string)
+    "same string"
+    "(Any (Var x) (Exists (Var y) (Equals (Var x) (Add (Var y) (Const 1)))))"
+    (Ast.string_of_formula (Parser.parse "AxEy x = y + 1"))
+
+let test_no_biggest_int () =
+  Alcotest.(check string)
+    "same string"
+    "(Not (Exists (Var x) (Any (Var y) (Exists (Var z) (Equals (Var x) (Add \
+     (Var y) (Var z)))))))"
+    (Ast.string_of_formula (Parser.parse "~ExAyEz(x = y + z)"))
+
+let test_divisible_by_7 () =
+  Alcotest.(check string)
+    "same string"
+    "(Exists (Var x) (Equals (Var y) (Add (Add (Add (Add (Add (Add (Var x) \
+     (Var x)) (Var x)) (Var x)) (Var x)) (Var x)) (Var x))))"
+    (Ast.string_of_formula (Parser.parse "Ex y = x + x + x + x + x + x + x"))
+
+let test_sum_of_evens_is_even () =
+  Alcotest.(check string)
+    "same string"
+    "(Any (Var x) (Any (Var y) (Impl (And (Exists (Var z) (Equals (Var x) (Add \
+     (Var z) (Var z)))) (Exists (Var w) (Equals (Var y) (Add (Var w) (Var \
+     w))))) (Exists (Var v) (Equals (Add (Var x) (Var y)) (Add (Var v) (Var \
+     v)))))))"
+    (Ast.string_of_formula
```

Comment:
And formatters can be used like this:
```suggestion
    (Format.asprintf "%a" Ast.format_formula
```

Reply:
I suggest ignoring this in order to make more good things rather than write "good-quality" code.